doc/contributor/README.md

```diff
@@ -0,0 +1,88 @@
+# Contributor Guides
+
+Thank you for contributing to `google-cloud-cpp`. We receive contributions from
+folks with very different backgrounds. Some of you are experienced C++
+developers, but this might be your first contribution to a GitHub hosted
+project. Some of your may be experienced GitHub contributors, but this might be
+the first time contributing to a C++ library project. Some of you may be
+experienced with both, but `google-cloud-cpp` has special requirements to
+support multiple platforms, compilers, and even multiple build tools.
+
+To help you navigate the project we have prepared (over time) a number of
+documents. You do not need to read each one, specially if they addressed areas
+you are already familiar with, but you may want to browse them in case something
+is surprising (or wrong! We need to improve these documents too).
+
+## Key Documents
+
+* [Forks and Pull Requests](howto-guide-forks-and-pull-requests.md): the basic
+  workflows for GitHub projects.
+* [Running CI builds locally](howto-guide-running-ci-builds-locally.md): how to
+  reproduce CI results locally, for faster edit -> build -> test cycles.
+* [Setup a development workstation](howto-guide-setup-development-workstation.md):
+  how to setup a Linux or Windows (workstation or VM) for `google-cloud-cpp`
+  development.
+* [Setup CMake Dependencies](howto-guide-setup-cmake-environment.md): how to
+  install the dependencies of `google-cloud-cpp` in `$HOME` for easier and
+  faster development with CMake.
+* [Working with Bazel and CMake](working-with-bazel-and-cmake.md): this project
+  can be compiled with CMake or Bazel. Always update the CMake project files
+  first, as these files are used to generate `*.bzl` files loaded by Bazel.
+  More details in the linked document.
+
+## Style
+
+This repository follows the [Google C++ Style Guide](
+https://google.github.io/styleguide/cppguide.html), with some additional
+constraints specified in the [style guide](/doc/cpp-style-guide.md).
+Please make sure your contributions adhere to the style guide.
+
+Many of these rules (but not all of them) are enforced using `clang-tidy(1)`.
+
+## Formatting
+
+The code in this project is formatted with `clang-format(1)`, and our CI builds
+will check that the code matches the format generated by this tool before
+accepting a pull request. Please configure your editor or IDE to use the Google
+style for indentation and other whitespace. If you need to reformat one or more
+files, you can simply run our formatting script:
+
+```console
+# Set NCPU to how many threads you wish to use
+$ env NCPU=4 CHECK_STYLE=yes ./ci/check-style.sh
+```
+
+Please be  advised that `clang-format` has been known to generate slightly
+different formatting in different versions. We use version 10 (the latest on
```

Comment:
I think this is version 11 now (Fedora 33). Either way I feel like this will get outdated, maybe something like "you can see the latest clang-format version required by checking..."

Reply:
I could not figure out an easy way to discover the current version.  We should probably fix that, but I left some pointers.